R/teal_data-extract.R

```diff
@@ -0,0 +1,38 @@
+#'
+#' @section Subsetting:
+#' `x[names]` subsets objects in `teal_data` environment and limit the code to the necessary needed to build limited
+#' objects.
+#'
+#' @param names (`character`) names of objects included in `teal_subset` to subset
```

Comment:
yes, but with `i` it works for `numeric` input, and we only support `character` input :)

Reply:
@llrs-roche oh yeah, but we don't want to introduce anything for numeric as `qenv` is basically an `environment` that is not subsettable for numeric as it does not have a natural order (like atomic vectors have).

```r
library(teal.code)
q <- qenv()
q <- eval_code(q, "a <- 1; b <- 2")
q@env[1]
# > Error in q@env[1] : object of type 'environment' is not subsettable
q@env[[1]]
# > Error in q@env[[1]] : wrong arguments for subsetting an environment
q@env[["a"]]
```

Numbers are fine for atomic vectors, or simple S3 objects that naturally use indexes, but I don't think it is the case for the environmemnt. We actually don't want it specifically as we only allow extraction by name, not by numeric index, so that you specifically need to provide the name of the object. With numeric input you can get a different element, depending on the current state of the object : )

Reply:
> Sorry, to raise a new thing, could you add a test for numeric subsetting and NA_character_? I know we get the Assertion error, but I want to make sure this is tested/documented for users.

No worries, good idea : )

Reply:
Here are extended tests 5f17af01828ed28f7d65bf37088972b6f619af1a